### PR TITLE
fix(security): harden isMultimodalInput guard + remove hardcoded AWS account ID

### LINKED ISSUE #278: TD-002: Unsafe Type Assertion in isMultimodalInput
## Summary
The `isMultimodalInput` type guard uses unsafe `as` type assertion without validation, allowing incorrect type narrowing and potential runtime errors.

## Root Cause
- Line 358 casts unknown to `MultimodalInput` without validation
- No checks for null, undefined, or primitives
- No array validation before accessing `.length`

## Fix
Replace unsafe cast with proper validation:
```typescript
export function isMultimodalInput(input: unknown): input is MultimodalInput {
  // Validate input is non-null object
  if (!input || typeof input !== "object") {
    return false;
  }
  
  const maybeInput = input as Record<string, unknown>;
  
  // Validate with Array.isArray()
  const hasImages = Array.isArray(maybeInput.images) && maybeInput.images.length > 0;
  // ... validate other properties
  
  return hasImages || hasCsvFiles || hasPdfFiles || ...;
}
```

## Security Risk Example
```typescript
const malformed = { images: "not-an-array" };
if (isMultimodalInput(malformed)) {
  // FALSE POSITIVE - Type guard incorrectly returns true
  malformed.images.forEach(img => process(img)); // CRASH!
}
```

## Acceptance Criteria
- [ ] Remove unsafe `as` casting
- [ ] Add null/undefined/primitive checks
- [ ] Use `Array.isArray()` for validation
- [ ] 20+ test cases covering edge cases
- [ ] All tests pass

## Files to Modify
- `src/lib/types/multimodal.ts` (lines 357-368)
- Create `test/types/multimodal-guards.test.ts`

### PR DESCRIPTION
**Pull request by murdore**
Fixes #278 #1012. isMultimodalInput required non-empty ARRAY per field (a string .length was truthy → false-positive + downstream crash); replaced hardcoded AWS account ID 225681119357 with placeholder 123456789012 (config.ts x2 + demo). Regression test; bugfixes 90/90.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation for multimodal inputs, rejecting malformed or empty fields and accepting only correctly structured content.
  * Updated Amazon Bedrock default model configuration to use the current inference profile.

* **Tests**
  * Added coverage for valid and invalid multimodal input formats.

### CODE REVIEW

**Review by Tara-ag — APPROVED** [review]
## Review Summary

Reviewed all 4 changed files in this PR:

1. `src/lib/types/multimodal.ts` — The `isMultimodalInput` guard correctly hardens the array-field checks using `Array.isArray(v) && v.length > 0`, fixing the string/number `.length` false-positive crash (#278). No new blocking issues.
2. `src/cli/commands/config.ts` — Hardcoded AWS account ID replaced with the standard placeholder. No new blocking issues.
3. `neurolink-demo/server.ts` — Same placeholder replacement. No new blocking issues.
4. `test/continuous-test-suite-bugfixes.ts` — Regression test added in the repo’s tsx-driven suite, consistent with CONTRIBUTING.md/CLAUDE.md testing conventions. No new blocking issues.

I did not raise any new inline comments; the points already noted by the earlier Copilot review (type-predicate soundness and remaining AWS ID occurrences) are already under discussion and I am not duplicating them.

**Decision:** Approve — no new blocking criteria met.

File: src/lib/types/multimodal.ts [hunk 1]
```diff
@@ -659,18 +659,26 @@ function isDirectorSegment(segment: unknown): segment is DirectorSegment {
 }
 
 export function isMultimodalInput(input: unknown): input is MultimodalInput {
-  const maybeInput = input as MultimodalInput;
+  if (typeof input !== "object" || input === null) {
+    return false;
+  }
+  const m = input as MultimodalInput;
```
Comment (Tara-ag) [inline src/lib/types/multimodal.ts:665]: ⚠️ **MAJOR**: The guard still casts `input as MultimodalInput` before validating fields. This type assertion undermines the hardening: TypeScript now treats `m.images` etc. as their declared array types even when the runtime value is a string/number, and the predicate becomes unsound for callers.

Additionally, `MultimodalInput` declares `text: string` as required, but the guard returns `true` for `{ images: [Buffer.from("x")] }` without `text`, so callers that narrow to `MultimodalInput` and access `.text` will get `undefined`.

Suggested fix:
```ts
export function isMultimodalInput(input: unknown): input is MultimodalInput {
  if (typeof input !== "object" || input === null) {
    return false;
  }
  const m = input as Record<string, unknown>;

  if (typeof m.text !== "string") {
    return false;
  }

  const isOptionalArray = (v: unknown): boolean =>
    v === undefined || Array.isArray(v);
  const isOptionalSegments = (v: unknown): boolean =>
    v === undefined || (Array.isArray(v) && v.every(isDirectorSegment));

  if (
    !isOptionalArray(m.images) ||
    !isOptionalArray(m.csvFiles) ||
    !isOptionalArray(m.pdfFiles) ||
    !isOptionalArray(m.files) ||
    !isOptionalArray(m.content) ||
    !isOptionalArray(m.audioFiles) ||
    !isOptionalArray(m.videoFiles) ||
    !isOptionalSegments(m.segments)
  ) {
    return false;
  }

  const hasItems = (v: unknown): boolean => Array.isArray(v) && v.length > 0;
  return (
    hasItems(m.images) ||
    hasItems(m.csvFiles) ||
    hasItems(m.pdfFiles) ||
    hasItems(m.files) ||
    hasItems(m.content) ||
    hasItems(m.audioFiles) ||
    hasItems(m.videoFiles) ||
    hasItems(m.segments)
  );
}
```

File: src/lib/types/multimodal.ts [hunk 2]
```diff
@@ -659,18 +659,26 @@ function isDirectorSegment(segment: unknown): segment is DirectorSegment {
...
+    hasItems(m.csvFiles) ||
+    hasItems(m.pdfFiles) ||
+    hasItems(m.files) ||
+    hasItems(m.content) ||
+    hasItems(m.audioFiles) ||
+    hasItems(m.videoFiles) ||
+    (Array.isArray(m.segments) &&
+      m.segments.length > 0 &&
+      m.segments.every(isDirectorSegment))
   );
```
Comment (Tara-ag) [inline src/lib/types/multimodal.ts:682]: ⚠️ **MAJOR**: Mixed valid/invalid fields produce a false-positive narrowing. For example, `{ images: "not-an-array", files: [Buffer.from("x")] }` returns `true` because `hasItems(m.files)` is true, but the narrowed type promises `images` is a valid array, so downstream code iterating `input.images` will crash.

Validate that every defined multimodal field has a valid type before checking for items. See the suggestion on line 665 for a committable fix.

File: test/continuous-test-suite-bugfixes.ts [hunk 3]
```diff
@@ -228,6 +229,23 @@ const tests: TestFunction[] = [
...
+        [{ csvFiles: 42 }, false],
+        [null, false],
+        ["hello", false],
+        [{}, false],
+        [{ images: [Buffer.from("x")] }, true],
+        [{ files: [{ buffer: Buffer.from("x"), filename: "a.pdf" }] }, true],
+      ];
+      return cases.every(([inp, want]) => isMultimodalInput(inp) === want);
+    },
+  },
```
Comment (Tara-ag) [inline test/continuous-test-suite-bugfixes.ts:248]: 💡 **MINOR**: The PR objective reportedly requires at least 20 edge-case test cases, but this `cases` array only has 7. The existing cases also miss mixed valid/invalid fields (e.g. `{ images: "bad", files: [Buffer] }`), empty arrays, and malformed `segments`. Consider expanding coverage so the guard's strict-validation behavior is pinned in CI.

Suggested additions:
```ts
[{ images: [] }, false],
[{ segments: [] }, false],
[{ images: "bad", files: [Buffer.from("x")] }, false],
[{ csvFiles: [Buffer.from("a")], pdfFiles: {} }, false],
[{ segments: [{ prompt: 123, image: "img" }] }, false],
[{ segments: [{ prompt: "test" }] }, false],
[{ images: ["img1"], pdfFiles: ["pdf1"] }, true],
[{ segments: [{ prompt: "test", image: "img" }] }, true],
```

File: src/cli/commands/config.ts [hunk 4]
```diff
@@ -56,7 +56,7 @@ const ConfigSchema: z.ZodType<CliNeuroLinkConfig> = z.object({
           model: z
             .string()
             .default(
-              "arn:aws:bedrock:us-east-2:225681119357:inference-profile/us.anthropic.claude-3-7-sonnet-20250219-v1:0",
+              "arn:aws:bedrock:us-east-2:123456789012:inference-profile/us.anthropic.claude-3-7-sonnet-20250219-v1:0",
             ),
```
Comment (Tara-ag) [inline src/cli/commands/config.ts:60]: 💡 **MINOR**: The hardcoded AWS account ID was replaced here, but the real account ID `225681119357` still appears elsewhere in the repo: `memory-bank/progress.md`, `.clinerules`, `memory-bank/archive/.clinerules-2025-01-07`, and `docs/cli-recordings/latest/provider-status-working.cast`. If the goal of issue #1012 is to scrub the real account ID from the repository, these additional occurrences should also be updated (or the PR scope should explicitly exclude documentation/memory-bank artifacts).

File: neurolink-demo/server.ts [hunk 5]
```diff
@@ -51,7 +51,7 @@ const ALL_PROVIDERS = [
 const DEFAULT_MODELS: Record<string, string> = {
```
Comment (Tara-ag) [inline neurolink-demo/server.ts:51]: 💡 **MINOR**: Same placeholder-account replacement as `src/cli/commands/config.ts`. This is consistent with the PR, but note that the real account ID remains in other files (memory-bank, .clinerules, cast recording). If #1012 intends full scrubbing, expand the cleanup to those artifacts or document the scope.

**Review by Tara-ag — APPROVED** [review]
Reviewed 4 files.

**New issues raised this run:** 1 MINOR — `neurolink-demo/server.ts` notes that the demo default ARN was updated, but the PR scope for scrubbing the real AWS account ID (#1012) should be clarified since the old ID still appears in `memory-bank/progress.md`, `.clinerules`, `memory-bank/archive/.clinerules-2025-01-07`, and `docs/cli-recordings/latest/provider-status-working.cast`.

**Duplicates of existing open comments:** The other inline observations repeat points already raised by `copilot-pull-request-reviewer` and `coderabbitai` (type-assertion unsoundness / missing `text` validation on `src/lib/types/multimodal.ts:665`, mixed-field false-positive narrowing on `:682`, insufficient edge-case tests on `test/continuous-test-suite-bugfixes.ts:248`, and remaining account-ID occurrences on `src/cli/commands/config.ts:60`). Per review rules, those are excluded from this decision.

**Decision:** No new blocking criteria are met — no hardcoded secrets introduced (the placeholder `123456789012` is the standard AWS dummy account), no critical security flaw, no public SDK breaking change, no CLAUDE.md critical-rule violation, and fewer than 3 new MAJOR issues. Approving.

Note: the pre-existing MAJOR comments on `src/lib/types/multimodal.ts` remain unresolved and should still be addressed before merge.
